chat_client/src/main.py

```diff
@@ -4,119 +4,186 @@
+        except ConnectionRefusedError as e:
             logging.error("Cannot connect [%s]", e)
+            self.disconnect()
         else:
             self._is_connected = True
             logging.info("Chat client connected")
 
-    def change_login_name(self) -> None:
-        """Change or set user login name"""
-        self._set_login_name()
+    def _handle_register(self) -> None:
+        """Handles user register, takes username, full name and password from user.
+        User have 3 chances to input valid password.
+
+        Raises:
+            rpc_error: Raised when error type was not expected.
+        """
+        if input(
+            "Do you want to register first? yes/[no]"
+        ).strip().lower() in ["y", "yes"]:
+            username = self._get_username()
+            full_name = input("Full name:").strip()
+            password = getpass()
+            req = chat_pb2.RegisterUserRequest(
+                user_info=chat_pb2.UserInfo(
+                    login=username, full_name=full_name
+                ),
+                password=password,
+            )
+            try:
+                self._stub.RegisterUser(request=req)
+            except grpc.RpcError as rpc_error:
+                if rpc_error.code() == grpc.StatusCode.ALREADY_EXISTS:
+                    logging.info("User %s already exists...", username)
+                elif rpc_error.code() == grpc.StatusCode.UNAVAILABLE:
+                    logging.debug("Server unavaible...")
+                else:
+                    raise rpc_error
+            else:
+                logging.info("User %s registered successfully", username)
+                return
+        else:
+            return
+
+    def _handle_login(self) -> None:
+        """Handles user login, user is asked to provide username and password.
+        User have 3 chances to input valid password.
+
+        Raises:
+            rpc_error: Raised when error type was not expected.
+            ConnectionRefusedError: Raised when user provide wrong creds 3 times.
+        """
+        username = self._get_username()
+        for _ in range(3):
+            password = getpass()
+            try:
+                self._stub.LoginUser(
+                    request=chat_pb2.LoginUserRequest(
+                        login=username, password=password
+                    )
+                )
+            except grpc.RpcError as rpc_error:
+                if rpc_error.code() == grpc.StatusCode.UNAUTHENTICATED:
+                    logging.info("Login failed, username: %s", username)
+                elif rpc_error.code() == grpc.StatusCode.UNAVAILABLE:
+                    logging.debug("Server unavaible...")
+                    break
+                else:
+                    raise rpc_error
+            else:
+                self._username = username
+                return
+        raise ConnectionRefusedError("Login failed")
+
+    def _get_username(self) -> str:
```

Comment:
Shouldn't it be rather `ask_username`? `Get` supposes that you wanna get smth, but in this case you literally asks

Reply:
Yes, you are right